### ISSUE TITLE
Extension host terminated unexpectedly. Please reload the window to recover.

### ISSUE THREAD

**Issue by netcore-jroger** [Microsoft/vscode#26628]
- VSCode Version: Code 1.12.1 (f6868fce3eeb16663840eb82123369dec6077a9b, 2017-05-04T21:26:50.689Z)
- OS Version: Windows_NT ia32 10.0.15063
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|npm-intellisense|christian-kohler|1.3.0|
|vscode-eslint|dbaeumer|1.2.8|
|githistory|donjayamanne|0.2.0|
|python|donjayamanne|0.6.4|
|tslint|eg2|0.15.0|
|csharpextensions|jchannon|1.3.0|
|docomment|k--kato|0.0.14|
|vue|liuji-jim|0.1.5|
|mssql|ms-mssql|1.0.0|
|csharp|ms-vscode|1.9.0|
|PowerShell|ms-vscode|1.0.0|
|vetur|octref|0.6.8|
|material-icon-theme|PKief|2.0.5|
|view-in-browser|qinjia|0.0.5|
|code-settings-sync|Shan|2.7.0|
|html-css-class-completion|Zignd|1.2.0|;

---

Steps to Reproduce:

1.Open the VSCode will show this error.

**Comment by bentolor** [comment]
Same here. Fresh install of VS 1.12.1 @ Win7. VS directly crashes after launch even if I call `Code.exe  --disable-extensions`

**Comment by netcore-jroger** [comment]
when I open dev tool. switch to `console` tab, find this error.
![qq 20170516105858](https://cloud.githubusercontent.com/assets/3223890/26087988/cc2cb170-3a26-11e7-9467-715639d77b60.png)

**Comment by netcore-jroger** [comment]
maximum 128 reports/day
reporter is electron-crash-service
[6188:22364:0516/110142.781:ERROR:crash_service.cc(295)] could not start dumper
[22920:20560:0516/110149.010:VERBOSE1:crash_service.cc(325)] client end. pid = 14748
[22920:19688:0516/110150.774:VERBOSE1:crash_service.cc(317)] client start. pid = 13484
[22920:19688:0516/110150.778:VERBOSE1:crash_service.cc(317)] client start. pid = 12880
[4840:18156:0516/110150.786:VERBOSE1:crash_service_main.cc(78)] Session start. cmdline is [--reporter-url=https://ticinocrashreporter.azurewebsites.net/crash --application-name=VSCode "--crashes-directory=C:\Users\xxxxx\AppData\Local\Temp\VSCode Crashes" --v=1]
[4840:18156:0516/110150.789:VERBOSE1:crash_service.cc(142)] window handle is 03BC093A
[4840:18156:0516/110150.789:VERBOSE1:crash_service.cc(284)] pipe name is \\.\pipe\VSCode Crash Service
dumps at C:\Users\xxxxx\AppData\Local\Temp\VSCode Crashes
[4840:18156:0516/110150.789:VERBOSE1:crash_service.cc(288)] checkpoint is C:\Users\xxxxx\AppData\Local\Temp\VSCode Crashes\crash_checkpoint.txt
server is https://ticinocrashreporter.azurewebsites.net/crash
maximum 128 reports/day
reporter is electron-crash-service
[4840:18156:0516/110150.789:ERROR:crash_service.cc(295)] could not start dumper
[17772:12484:0516/110150.791:VERBOSE1:crash_service_main.cc(78)] Session start. cmdline is [--reporter-url=https://ticinocrashreporter.azurewebsites.net/crash --application-name=VSCode "--crashes-directory=C:\Users\xxxxx\AppData\Local\Temp\VSCode Crashes" --v=1]
[17772:12484:0516/110150.794:VERBOSE1:crash_service.cc(142)] window handle is 001009F4
[17772:12484:0516/110150.794:VERBOSE1:crash_service.cc(284)] pipe name is \\.\pipe\VSCode Crash Service
dumps at C:\Users\xxxxx\AppData\Local\Temp\VSCode Crashes
[17772:12484:0516/110150.794:VERBOSE1:crash_service.cc(288)] checkpoint is C:\Users\xxxxxx\AppData\Local\Temp\VSCode Crashes\crash_checkpoint.txt
server is https://ticinocrashreporter.azurewebsites.net/crash
maximum 128 reports/day
reporter is electron-crash-service
[17772:12484:0516/110150.794:ERROR:crash_service.cc(295)] could not start dumper
[22920:20560:0516/110201.838:VERBOSE1:crash_service.cc(325)] client end. pid = 13484
[22920:20560:0516/110202.159:VERBOSE1:crash_service.cc(325)] client end. pid = 12880
[22920:20560:0516/110202.159:VERBOSE1:crash_service.cc(325)] client end. pid = 12880
[22920:20560:0516/110203.160:VERBOSE1:crash_service.cc(346)] zero clients. exiting
[22920:16980:0516/110203.160:VERBOSE1:crash_service.cc(486)] session ending..
[22920:16980:0516/110203.160:VERBOSE1:crash_service.cc(491)] clients connected :4
clients terminated :4
dumps serviced :0
dumps reported :0
[22920:16980:0516/110203.160:VERBOSE1:crash_service_main.cc(99)] Session end. return code is 0

**Comment by joaomoreno** [comment]
Can you try out our [insider release](https://code.visualstudio.com/insiders)?

**Comment by OYMiss** [comment]
When I tried the insider release? 
It still have the error.
I`m using windows 10.
And when I disable the Python extension ,the error disappeared.

**Comment by bentolor** [comment]
@joaomoreno Mends it for me.

**Comment by netcore-jroger** [comment]
When I tried the insider release.
It still have the error.

**Comment by netcore-jroger** [comment]
What information do I need to provide?

**Comment by joaomoreno** [comment]
@netcore-jroger Did you see @OYMiss's comment?

> And when I disable the Python extension ,the error disappeared.

Is it true also for you?

**Comment by netcore-jroger** [comment]
@joaomoreno now I have nothing plugins in VSCode.The error always exist.

**Comment by netcore-jroger** [comment]
any body Still need me to provide any information?

**Comment by joaomoreno** [comment]
If you disable git `git.enabled = false`, does the error still occur?

**Comment by netcore-jroger** [comment]
@joaomoreno yes!

**Comment by joaomoreno** [comment]
That's good. @bpasero any ideas?

**Comment by bpasero** [comment]
Here is a **checklist** to find out more about the freeze/slow/crash
- [ ] it reproduces with extensions disabled (`code --disable-extensions` from the command line - NOT the integrated terminal of Code)
    - if it does not reproduce with extensions disabled, you can stop testing and report this!
- [ ] it was not happening in the previous stable release
- [ ] it reproduces in insiders builds (get it from [here](http://code.visualstudio.com/Download#insiders))
- [ ] there is no suspicious output when running from the command line (`code --verbose`)
- [ ] there is no suspicious output in the window itself (Help | Toggle Developer Tools | Console)
- [ ] it reproduces on Windows, Linux and Mac
- [ ] it reproduces after deleting `<appdata>/User/keybindings.json` and `<appdata>/User/settings.json` (backup first! see below for paths)
- [ ] it reproduces when opening empty (`code -n` from the command line)
- [ ] it reproduces opening any folder
- [ ] it reproduces opening any file
- [ ] it reproduces when only having a single window open
- [ ] it reproduces by just using Code without external tools running on the same folder
- [ ] it reproduces from a workspace that is not under Git version control
- [ ] it reproduces independently if a file is opened or not after startup
- [ ] it reproduces when running with GPU disabled (`code --disable-gpu`)
- [ ] it reproduces when being offline
- [ ] it reproduces when disabling the file watching (set `files.watcherExclude` to `"**/**": true`)
- [ ] it reproduces after uninstalling and reinstalling
- [ ] it reproduces after deleting the application data directory (backup first! see below for paths)
- [ ] I have not played around with permissions or ran Code as administrator
- [ ] I am seeing a single CPU process spin high (please share the full command line arguments)
- [ ] I am seeing Code consuming lots of memory in a short time

**Application Data Directory for VS Code:**
* Windows: `C:\Users\<user>\AppData\Roaming\Code`
* Mac: `/Users/<user>/Library/Application Support/Code`
* Linux: `/home/<user>/.config/Code`

**Application Data Directory for VS Code Insiders:**
* Windows: `C:\Users\<user>\AppData\Roaming\Code - Insiders`
* Mac: `/Users/<user>/Library/Application Support/Code - Insiders`
* Linux: `/home/<user>/.config/Code - Insiders`

**Comment by netcore-jroger** [comment]
use command line.  code.exe --verbose
![verbose](https://cloud.githubusercontent.com/assets/3223890/26434022/c602dc84-4138-11e7-873d-6beeacb24444.png)

**Comment by netcore-jroger** [comment]
![verbose-1](https://cloud.githubusercontent.com/assets/3223890/26434102/306c8b2e-4139-11e7-93fa-8f22434975d0.png)

**Comment by netcore-jroger** [comment]
![disable-extensions](https://cloud.githubusercontent.com/assets/3223890/26434224/03484a92-413a-11e7-9afc-112b6f7b33e6.png)

**Comment by netcore-jroger** [comment]
![disable-gpu](https://cloud.githubusercontent.com/assets/3223890/26434289/8e60735c-413a-11e7-8553-fa4eaf1ac1d2.png)

**Comment by netcore-jroger** [comment]
![filewatchers](https://cloud.githubusercontent.com/assets/3223890/26434362/f4f3ec7a-413a-11e7-8130-b4bdc1851eff.png)

**Comment by netcore-jroger** [comment]
toggle dev tool  still get same error.
![same-error](https://cloud.githubusercontent.com/assets/3223890/26434429/4febd7dc-413b-11e7-9bc7-f19ce3cecd75.png)

**Comment by netcore-jroger** [comment]
memory cpu
![memory-cpu](https://cloud.githubusercontent.com/assets/3223890/26434651/6c96b5b8-413c-11e7-8f4e-abb63ea6e0a2.gif)

**Comment by netcore-jroger** [comment]
ran Code as administrator still get same error.

**Comment by bpasero** [comment]
@netcore-jroger strange, thanks for going through the list. I wonder if this is locale, keyboard and environment related.

Does it matter if you start from the command line vs. from the dock?

**Comment by netcore-jroger** [comment]
@bpasero same error.

**Comment by bpasero** [comment]
@netcore-jroger does it reproduce if you startup stable on a different data directory: `code --user-data-dir <some folder>`?

**Comment by netcore-jroger** [comment]
ok, let me try.

**Comment by netcore-jroger** [comment]
**same error.**

![qq 20170526141251](https://cloud.githubusercontent.com/assets/3223890/26483033/76915c62-421d-11e7-8df7-34b08ac5e451.png)

**Comment by whattheken** [comment]
I have the same issue, and it seems to happen if running visual studio code as an non-admin windows user (but worked as non-admin in previous versions). This does not occur as an admin user.

Went through this checklist below:

- [x] it reproduces with extensions disabled (code --disable-extensions from the command line - NOT the integrated terminal of Code)
- [x] if it does not reproduce with extensions disabled, you can stop testing and report this!
- [x] it was not happening in the previous stable release
- [x] it reproduces in insiders builds (get it from here)
- [x] there is no suspicious output when running from the command line (code `--verbose
```
[main 9:02:26 PM] args: { _: [],
  help: false,
  h: false,
  version: false,
  v: false,
  wait: false,
  w: false,
  diff: false,
  d: false,
  goto: false,
  g: false,
  'new-window': false,
  n: false,
  'unity-launch': false,
  'reuse-window': false,
  r: false,
  performance: false,
  p: false,
  'prof-startup': false,
  verbose: true,
  logExtensionHostCommunication: false,
  'disable-extensions': false,
  disableExtensions: false,
  'list-extensions': false,
  'show-versions': false,
  nolazy: false }
[main 9:02:27 PM] IPC#vscode-machineId
[main 9:02:27 PM] IPC#vscode-workbenchLoaded
```
- [x] there is no suspicious output in the window itself (Help | Toggle Developer Tools | Console)
- [] it reproduces on Windows, Linux and Mac - COMMENT: only tried on Windows 10
- [x]  it reproduces after deleting <appdata>/User/keybindings.json and <appdata>/User/settings.json (backup first! see below for paths)
- [x]  it reproduces when opening empty (code -n from the command line)
- [x]  it reproduces opening any folder
- [x]  it reproduces opening any file
- [x]  it reproduces when only having a single window open
- [x]  it reproduces by just using Code without external tools running on the same folder
- [x]  it reproduces from a workspace that is not under Git version control
- [x]  it reproduces independently if a file is opened or not after startup
- [x]  it reproduces when running with GPU disabled (code --disable-gpu)
- [x]  it reproduces when being offline
- [x]  it reproduces when disabling the file watching (set files.watcherExclude to "**/**": true)
- [x]  it reproduces after uninstalling and reinstalling
- [x]  it reproduces after deleting the application data directory (backup first! see below for paths)
- []  I have not played around with permissions or ran Code as administrator - COMMENT: ran as admin after non-admin user not working, admin user works without issue
- []  I am seeing a single CPU process spin high (please share the full command line arguments) - COMMENT: not seeing high cpu process
- []  I am seeing Code consuming lots of memory in a short time - COMMENT: Code not consuming lots of memory

Removing all folders in C:\Program Files (x86)\Microsoft VS Code Insiders\resources\app\extensions causes the error to go away, but this can't be the right solution?

**Comment by netcore-jroger** [comment]
I upgrade to v1.13.0 also has the issue.

**Comment by netcore-jroger** [comment]
I try deleted `D:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\git` folder.  , then this issue has resolved. My god.

**Comment by netcore-jroger** [comment]
suggestion: The mechanism of using the plugin is independent of each other, and the plugin does not affect each other.

**Comment by netcore-jroger** [comment]
so I think the `--disable-extensions` argument does not work.

**Comment by netcore-jroger** [comment]
@bpasero see [Find the logic of git](https://github.com/Microsoft/vscode/blob/master/extensions/git/src/git.ts#L130-L136). If git does not install to C drive , then can not get the path to git.exe. Now has no problem when I start vscode.exe, but if I press F5 debugging node.js procedures, there will be this problem.

**Comment by netcore-jroger** [comment]
![git_error](https://user-images.githubusercontent.com/3223890/27362988-56054c14-5664-11e7-8f90-9f6736864210.png)
why not find the path to git.exe from system- environment-variables ?

**Comment by leitingx762** [comment]
Ouch...Thanks for you!  @netcore-jroger  I deleted this folder. The error has been repaired.

**Comment by Zaseth** [comment]
This also worked for me. Thank you very much! (using Insiders edition)

**Comment by netcore-jroger** [comment]
In fact, the root cause of my problem is that the environmental variables of the system are contaminated. Please check here for specific reasons [nodejs/help#728](https://github.com/nodejs/help/issues/728)

**Comment by Zaseth** [comment]
Still getting the error on PHP files. Any idea?

**Comment by whattheken** [comment]
For me, this seems to be an issue with getting the git path for non-admin Windows user. Here's what happens for me on Windows 10.

Scenario 1: git path set in non-admin user environment variable
- running VS code as non-admin user - extension host error
- running VS code as admin user - no error, but no git

Scenario 2: git path set in admin user environment variable
- running VS code as non-admin user results in extension host error
- running VS code as admin user - no error, and git works

Scenario 3: git path set as system environment variable
- running VS code as non-admin user - no error
- running VS code as admin user - no error

Seems like VS code has problems reading non-admin user environment variables? In a non-admin user account, running "git --version" returns the valid version when the git path is set on the non-admin user, but seems like maybe something to do with this git logic [here](https://github.com/Microsoft/vscode/blob/master/extensions/git/src/git.ts#L59-L67)?

**Comment by marcusrs** [comment]
I don't know if this will help someone but I decided to uninstalled and reinstalled vs code 1.15.1 stable. Still got the same error, then I decided to install vs code insider version instead of the stable version and still got the same error. I disable all the extensions and went through them one by one. Until I came upon extension Typescript Hero. I reloaded this extension and got the error. I uninstalled it and have not have that error sense. Then it pop up again and I notice in the developer tool that there was a lot of conflict between all the extensions I installed. So I went through and started removing all the ones that were conflicting with each other.   I think that with vs code it is best to stick with the bare minimum of extensions needed for your project..

**Comment by cihanbas** [comment]
@netcore-jroger Very very thsnk you for answer 👍

**Comment by meraj-shaikh** [comment]
Had same issue for PHP files, by setting below values in user settings solved the error pop up.
"php.suggest.basic": false,
    "php.validate.enable": false

**Comment by Alejandro-MartinG** [comment]
I had the same problem with a clean installation, the solution working for me was proposed by @netcore-jroger. Thank you so much.
(I am using the last version 1.17 and windows 10)

> https://github.com/Microsoft/vscode/issues/26628#issuecomment-307709404

**Comment by jdc-cunningham** [comment]
I'm not sure if this is what caused the error for me (regarding extension host terminated not going away) I had an incorrect settings line in JSON pertaining to python linting. I couldn't even load a theme. I turned off the php suggest/validate as @meraj-shaikh mentioned, that worked for me, but I also removed that python linting line that I added at some point. After I removed that line I was able to set my theme again.

Prior to this I deleted the extensions folder under .vscode but it kept coming back even after uninstall on Windows 10.
